src/PhpUnit/UriInterfaceTestsTrait.php

```diff
@@ -0,0 +1,290 @@
+<?php
+
+namespace Maks3w\Psr7Assertions\PhpUnit;
+
+use PHPUnit_Framework_Assert as Assert;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Provide PHPUnit test methods for Psr\Http\Message\UriInterface constraints.
+ */
+trait UriInterfaceTestsTrait
+{
+    /**
+     * @return Psr\Http\Message\UriInterface
+     */
+    abstract public function createUriObject($url);
+
+    public function testUriImplementsInterface()
+    {
+        Assert::assertInstanceOf('Psr\Http\Message\UriInterface', $this->createUriObject(''));
+    }
+
+    /**
+     * @dataProvider uriProvider
+     * @group scheme
```

Comment:
I don't see useful `@group` tags here. Seems you want to group by method under test but the name choosen potentially conflicts with other methods with the same name in the rest of interfaces.

Aditionally the test function name is already explicit about the method name so it's simple to execute only the desired test with phpunit's filter argument.